
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Pipelines overview &#8212; samgen  documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Running pipelines" href="../tech_docs/tech_docs.html" />
    <link rel="prev" title="System overview" href="../overview/overview.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="pipelines-overview">
<h1>Pipelines overview<a class="headerlink" href="#pipelines-overview" title="Permalink to this headline">¶</a></h1>
<section id="aux-render-notebook">
<h2>aux-render-notebook<a class="headerlink" href="#aux-render-notebook" title="Permalink to this headline">¶</a></h2>
<img alt="../_images/aux-render-notebook.png" src="../_images/aux-render-notebook.png" />
<section id="description">
<h3>Description<a class="headerlink" href="#description" title="Permalink to this headline">¶</a></h3>
<p>This DAG is used to render a single Jupyter notebook for the
analysis of the results of the previously executed pipelines.
In order to run the DAG you should provide the following parameters:</p>
<section id="inputs-common">
<h4>Inputs common:<a class="headerlink" href="#inputs-common" title="Permalink to this headline">¶</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="s2">&quot;sample_id&quot;</span><span class="p">:</span> <span class="n">ID_of_the_sample</span><span class="p">,</span>
<span class="s2">&quot;run_id&quot;</span><span class="p">:</span> <span class="n">ID_of_the_executed_run</span><span class="p">,</span>
<span class="s2">&quot;dag_id&quot;</span><span class="p">:</span> <span class="n">ID_of_the_executed_DAG</span><span class="p">,</span>
<span class="s2">&quot;git_notebooks_version&quot;</span><span class="p">:</span> <span class="n">release_version_of_the_notebooks_repository</span>
</pre></div>
</div>
</section>
</section>
<section id="input-parameters">
<h3>Input parameters<a class="headerlink" href="#input-parameters" title="Permalink to this headline">¶</a></h3>
<p><cite>sample_input.json</cite></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;sample_id&quot;</span><span class="p">:</span> <span class="s2">&quot;germline_test_SID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;run_id&quot;</span><span class="p">:</span> <span class="s2">&quot;manual__2022-01-11T00_13_05.687947_00_00&quot;</span><span class="p">,</span>
  <span class="s2">&quot;dag_id&quot;</span><span class="p">:</span> <span class="s2">&quot;germline-exome&quot;</span><span class="p">,</span>
  <span class="s2">&quot;git_notebooks_version&quot;</span><span class="p">:</span> <span class="s2">&quot;0.1.11&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="sample-rest-api-call">
<h3>Sample REST API call<a class="headerlink" href="#sample-rest-api-call" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -v -g -X POST <span class="se">\</span>
  http://localhost:8080/api/v1/dags/aux-render-notebook/dagRuns <span class="se">\</span>
  -H <span class="s1">&#39;Cache-Control: no-cache&#39;</span> <span class="se">\</span>
  -H <span class="s1">&#39;Content-Type: application/json&#39;</span> <span class="se">\</span>
  -d <span class="s2">&quot;{&#39;conf&#39;:\&#39;</span><span class="k">$(</span>cat sample_input.json <span class="p">|</span> sed <span class="s1">&#39;s/\&quot;/\\&quot;/g&#39;</span><span class="k">)</span><span class="s2">\&#39;}&quot;</span>
</pre></div>
</div>
</section>
<section id="sample-airflow-cli-invocation">
<h3>Sample Airflow cli invocation<a class="headerlink" href="#sample-airflow-cli-invocation" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>airflow dags trigger <span class="se">\</span>
-c <span class="s2">&quot;</span><span class="k">$(</span>cat sample_input.json<span class="k">)</span><span class="s2">&quot;</span> <span class="se">\</span>
-r dag_run-1 aux-render-notebook
</pre></div>
</div>
</section>
</section>
<section id="germline-exome">
<h2>germline-exome<a class="headerlink" href="#germline-exome" title="Permalink to this headline">¶</a></h2>
<img alt="../_images/germline-exome.png" src="../_images/germline-exome.png" />
<section id="id1">
<h3>Description<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>Germline exome workflow based on <a class="reference external" href="https://github.com/sysamgen/analysis-workflows/blob/master/definitions/pipelines/germline_exome.cwl">mgi-germline-exome</a>.</p>
<p>This worfklow completes exome alignment and germline variant detection using a variety of tools.
The initial processing of sequence is performed by picard, alignment is performed by bwa. Output bam file undergoes qc process resulting in
insert_size_metrics, alignment_summary_metrics, hs_metrics and flagstat. VerifyBamId step checks if reads are not contaminated as a mixture of two samples.
Variants are detected using GATK HaplotypeCaller tool.</p>
<p>Germline pipeline starts with either fastq files or unaligned BAM files with necessary metadata describing analysis type, sample name and phenotype.</p>
<img alt="../_images/alignment_qc.png" src="../_images/alignment_qc.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1093/bioinformatics/btp324">bwa_mem 0.7.15</a> - for sequence alignment step using GRCh38 reference</p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu314">samblaster 0.1.24</a> - for adding mate tags</p></li>
<li><p><a class="reference external" href="10.3390/biology1030895">flexbar 3.5</a> - for bar and adapter removal</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard:2.23.6</a> for alignment metrics</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btp352">samtools 1.11–h6270b1f_0</a> for BAM indexing and flagstat operation</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard 2.18.1</a> - for bedtointervals operation</p></li>
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk 4.1.8.1</a>  for Select Variants</p></li>
<li><p><a class="reference external" href="doi:10.1016/j.ajhg.2012.09.004">verifyBamId v1.1.3</a> - for checking whether the reads are contaminated as a mixture of two samples</p></li>
<li><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a> - for bam QC report</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btv566">qualimap</a> - for bam QC report</p></li>
</ul>
<img alt="../_images/detect_variants.png" src="../_images/detect_variants.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk:4.1.8.1</a> Haplotype Caller for variant (SNPs and indels) detection</p></li>
<li><p><a class="reference external" href="10.1093/gigascience/giab008">bcftools</a> - QC for detected variants</p></li>
</ul>
<blockquote>
<div><p>After variants are detected by Haplotype Caller they are subject  decomposition and normalization to assure coherent variant representation.
Once variants are standardized they are annotated with metadata collected and published by public databases.
In the afterburner step final postprocessing is performed including generating URLs to external databases and extracting genotype information from INFO field.</p>
<p>Used tools:</p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btv112">vt 0.57721–hf74b74d_1</a> - for normalization and decomposition</p></li>
<li><p><a class="reference external" href="doi:10.1186/s13059-016-0974-4">vep 111</a> - for variants annotations</p></li>
</ul>
<p>After succesfull completion of variants annotations data are transformed into tabular form and stored in relational database using custom operations.</p>
</section>
<section id="id3">
<h3>Input parameters<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p><cite>sample_input.json</cite></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;design&quot;</span><span class="p">:</span> <span class="s2">&quot;test_design&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fastq1_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/germline-exome_tests/RG-corriell_S7_L001_R1_001.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fastq2_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/germline-exome_tests/RG-corriell_S7_L001_R2_001.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;flowcell_id&quot;</span><span class="p">:</span> <span class="s2">&quot;JHDAXCKJDLSK&quot;</span><span class="p">,</span>
  <span class="s2">&quot;gender&quot;</span><span class="p">:</span> <span class="s2">&quot;M&quot;</span><span class="p">,</span>
  <span class="s2">&quot;is_test&quot;</span><span class="p">:</span> <span class="s2">&quot;_tests&quot;</span><span class="p">,</span>
  <span class="s2">&quot;patient_id&quot;</span><span class="p">:</span> <span class="s2">&quot;germline_test_PID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;phenotype&quot;</span><span class="p">:</span> <span class="s2">&quot;proband&quot;</span><span class="p">,</span>
  <span class="s2">&quot;sample_id&quot;</span><span class="p">:</span> <span class="s2">&quot;germline_test_SID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;template_path&quot;</span><span class="p">:</span> <span class="s2">&quot;dags/cwl/configurations/templates/germline_exome_configuration_template.yaml&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="id4">
<h3>Sample REST API call<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -v -g -X POST <span class="se">\</span>
  http://localhost:8080/api/v1/dags/germline-exome/dagRuns <span class="se">\</span>
  -H <span class="s1">&#39;Cache-Control: no-cache&#39;</span> <span class="se">\</span>
  -H <span class="s1">&#39;Content-Type: application/json&#39;</span> <span class="se">\</span>
  -d <span class="s2">&quot;{&#39;conf&#39;:\&#39;</span><span class="k">$(</span>cat sample_input.json <span class="p">|</span> sed <span class="s1">&#39;s/\&quot;/\\&quot;/g&#39;</span><span class="k">)</span><span class="s2">\&#39;}&quot;</span>
</pre></div>
</div>
</section>
<section id="id5">
<h3>Sample Airflow cli invocation<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>airflow dags trigger <span class="se">\</span>
-c <span class="s2">&quot;</span><span class="k">$(</span>cat sample_input.json<span class="k">)</span><span class="s2">&quot;</span> <span class="se">\</span>
-r dag_run-1 germline-exome
</pre></div>
</div>
</section>
</section>
<section id="kf-rnaseq">
<h2>kf_rnaseq<a class="headerlink" href="#kf-rnaseq" title="Permalink to this headline">¶</a></h2>
<img alt="../_images/kf_rnaseq.png" src="../_images/kf_rnaseq.png" />
<section id="id6">
<h3>Description<a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h3>
<p>DAG = kf_rnaseq</p>
<p>RNA-seq Workflow based on <a class="reference external" href="https://github.com/kids-first/kf-rnaseq-workflow">kf-rnaseq-workflow</a>.</p>
<p>This pipeline utilizes cutadapt to trim adapters from the raw reads, if necessary, and passes the reads to STAR for alignment.
The alignment output is used by RSEM for gene expression abundance estimation.
Additionally, Kallisto is used for quantification, but uses pseudoalignments to estimate the gene abundance from the raw data.
Fusion calling is performed using Arriba and STAR-Fusion detection tools on the STAR alignment outputs.
Filtering and prioritization of fusion calls is done by annoFuse.
Metrics for the workflow are generated by RNA-SeQC.</p>
<p><a class="reference external" href="https://github.com/marcelm/cutadapt">Cutadapt v2.5</a> Cut adapter sequences from raw reads if needed.</p>
<p><a class="reference external" href="https://doi.org/f4h523">STAR v2.6.1d</a> RNA-Seq raw data alignment.</p>
<p><a class="reference external" href="https://doi:10/cwg8n5">RSEM v1.3.1</a> Calculation of gene expression.</p>
<p><a class="reference external" href="https://doi:10.1038/nbt.3519">Kallisto v0.43.1</a> Raw data pseudoalignment to estimate gene abundance.</p>
<p><a class="reference external" href="https://doi:10.1101/120295">STAR-Fusion v1.5.0</a> Fusion detection for <code class="docutils literal notranslate"><span class="pre">STAR</span></code> chimeric reads.</p>
<p><a class="reference external" href="https://github.com/suhrig/arriba/">Arriba v1.1.0</a> Fusion caller that uses <code class="docutils literal notranslate"><span class="pre">STAR</span></code> aligned reads and chimeric reads output.</p>
<p><a class="reference external" href="https://github.com/d3b-center/annoFuse/releases/tag/v0.90.0">annoFuse 0.90.0</a> Filter and prioritize fusion calls. For more information, please see the following <a class="reference external" href="https://www.biorxiv.org/content/10.1101/839738v3">paper</a>.</p>
<p><a class="reference external" href="https://github.com/broadinstitute/rnaseqc">RNA-SeQC v2.3.4</a> Generate metrics such as gene and transcript counts, sense/antisene mapping, mapping rates, etc</p>
<section id="id7">
<h4>Inputs common:<a class="headerlink" href="#id7" title="Permalink to this headline">¶</a></h4>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span>inputs:
  sample_name: string
  r1_adapter:  [&#39;null&#39;, string]
  r2_adapter: [&#39;null&#39;, string]
  STAR_outSAMattrRGline: string
  STARgenome: File
  RSEMgenome: File
  reference_fasta: File
  gtf_anno: File
  FusionGenome: File
  runThread: int
  RNAseQC_GTF: File
  kallisto_idx: File
  wf_strand_param: [&quot;default&quot;, &quot;rf-stranded&quot;, &quot;fr-stranded&quot;], doc: &quot;use &#39;default&#39; for unstranded/auto, &#39;rf-stranded&#39; if read1 in the fastq read pairs is reverse complement to the transcript, &#39;fr-stranded&#39; if read1 same sense as transcript&quot;
  input_type: [&quot;BAM&quot;, &quot;FASTQ&quot;], doc: &quot;Please select one option for input file type, BAM or FASTQ.&quot;
</pre></div>
</div>
</section>
<section id="bam-input-specific">
<h4>Bam input-specific:<a class="headerlink" href="#bam-input-specific" title="Permalink to this headline">¶</a></h4>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">inputs</span><span class="p">:</span>
  <span class="nt">reads1</span><span class="p">:</span> <span class="l l-Scalar l-Scalar-Plain">File</span>
</pre></div>
</div>
</section>
<section id="pe-fastq-input-specific">
<h4>PE Fastq input-specific:<a class="headerlink" href="#pe-fastq-input-specific" title="Permalink to this headline">¶</a></h4>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">inputs</span><span class="p">:</span>
  <span class="nt">reads1</span><span class="p">:</span> <span class="l l-Scalar l-Scalar-Plain">File</span>
  <span class="nt">reads2</span><span class="p">:</span> <span class="l l-Scalar l-Scalar-Plain">File</span>
</pre></div>
</div>
</section>
<section id="se-fastq-input-specific">
<h4>SE Fastq input-specific:<a class="headerlink" href="#se-fastq-input-specific" title="Permalink to this headline">¶</a></h4>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">inputs</span><span class="p">:</span>
  <span class="nt">reads1</span><span class="p">:</span> <span class="l l-Scalar l-Scalar-Plain">File</span>
</pre></div>
</div>
</section>
<section id="run">
<h4>Run:<a class="headerlink" href="#run" title="Permalink to this headline">¶</a></h4>
<p>1) Fastq or bam input are accepted:
For PE fastq input, please enter the reads 1 file in <code class="docutils literal notranslate"><span class="pre">reads1</span></code> and the reads 2 file in <code class="docutils literal notranslate"><span class="pre">reads2</span></code>.
For SE fastq input, enter the single ends reads file in <code class="docutils literal notranslate"><span class="pre">reads1</span></code> and leave <code class="docutils literal notranslate"><span class="pre">reads2</span></code> empty as it is optional.
For bam input, please enter the reads file in <code class="docutils literal notranslate"><span class="pre">reads1</span></code> and leave <code class="docutils literal notranslate"><span class="pre">reads2</span></code> empty as it is optional.</p>
<p>2) <code class="docutils literal notranslate"><span class="pre">r1_adapter</span></code> and <code class="docutils literal notranslate"><span class="pre">r2_adapter</span></code> are OPTIONAL.
If the input reads have already been trimmed, leave these as null and cutadapt step will simple pass on the fastq files to STAR.
If they do need trimming, supply the adapters and the cutadapt step will trim, and pass trimmed fastqs along.</p>
<ol class="arabic simple" start="3">
<li><p><code class="docutils literal notranslate"><span class="pre">wf_strand_param</span></code> is a workflow convenience param so that, if you input the following, the equivalent will propagate to the four tools that use that parameter:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>- `default`: &#39;rsem_std&#39;: null, &#39;kallisto_std&#39;: null, &#39;rnaseqc_std&#39;: null, &#39;arriba_std&#39;: null. This means unstranded or auto in the case of arriba.
- `rf-stranded`: &#39;rsem_std&#39;: 0, &#39;kallisto_std&#39;: &#39;rf-stranded&#39;, &#39;rnaseqc_std&#39;: &#39;rf&#39;, &#39;arriba_std&#39;: &#39;reverse&#39;.  This means if read1 in the input fastq/bam is reverse complement to the transcript that it maps to.
- `fr-stranded`: &#39;rsem_std&#39;: 1, &#39;kallisto_std&#39;: &#39;fr-stranded&#39;, &#39;rnaseqc_std&#39;: &#39;fr&#39;, &#39;arriba_std&#39;: &#39;yes&#39;. This means if read1 in the input fastq/bam is the same sense (maps 5&#39; to 3&#39;) to the transcript that it maps to.
</pre></div>
</div>
<ol class="arabic simple" start="4">
<li><p>Suggested <code class="docutils literal notranslate"><span class="pre">STAR_outSAMattrRGline</span></code>, with <strong>TABS SEPARATING THE TAGS</strong>,  format is:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>`ID:sample_name LB:aliquot_id   PL:platform SM:BSID` for example `ID:7316-242   LB:750189 PL:ILLUMINA SM:BS_W72364MN`
</pre></div>
</div>
<ol class="arabic simple" start="5">
<li><p>Suggested inputs are:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>- `FusionGenome`: [GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz](https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/__genome_libs_StarFv1.3/GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz)
- `gtf_anno`: gencode.v27.primary_assembly.annotation.gtf, location: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.primary_assembly.annotation.gtf.gz, will need to unzip
- `RNAseQC_GTF`: gencode.v27.primary_assembly.RNAseQC.gtf, built using `gtf_anno` and following build instructions [here](https://github.com/broadinstitute/rnaseqc#usage)
- `RSEMgenome`: RSEM_GENCODE27.tar.gz, built using the `reference_fasta` and `gtf_anno`, following `GENCODE` instructions from [here](https://deweylab.github.io/RSEM/README.html), then creating a tar ball of the results.
- `STARgenome`: STAR_GENCODE27.tar.gz, created using the star_genomegenerate.cwl tool, using the `reference_fasta`, `gtf_anno`, and setting `sjdbOverhang` to 100
- `reference_fasta`: [GRCh38.primary_assembly.genome.fa](ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/GRCh38.primary_assembly.genome.fa.gz), will need to unzip
- `kallisto_idx`: gencode.v27.kallisto.index, built from gencode 27 trascript fasta: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.transcripts.fa.gz, following instructions from [here](https://pachterlab.github.io/kallisto/manual)
</pre></div>
</div>
</section>
<section id="outputs">
<h4>Outputs:<a class="headerlink" href="#outputs" title="Permalink to this headline">¶</a></h4>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span>outputs:
  cutadapt_stats: File, outputSource: cutadapt/cutadapt_stats # only if adapter supplied
  STAR_transcriptome_bam: File, outputSource: star/transcriptome_bam_out
  STAR_sorted_genomic_bam: File, outputSource: samtools_sort/sorted_bam
  STAR_sorted_genomic_bai: File, outputSource: samtools_sort/sorted_bai
  STAR_chimeric_bam_out: File, outputSource: samtools_sort/chimeric_bam_out
  STAR_chimeric_junctions: File, outputSource: star_fusion/chimeric_junction_compressed
  STAR_gene_count: File, outputSource: star/gene_counts
  STAR_junctions_out: File, outputSource: star/junctions_out
  STAR_final_log: File, outputSource: star/log_final_out
  STAR-Fusion_results: File, outputSource: star_fusion/abridged_coding
  arriba_fusion_results: File, outputSource: arriba_fusion/arriba_fusions
  arriba_fusion_viz: File, outputSource: arriba_fusion/arriba_pdf
  RSEM_isoform: File, outputSource: rsem/isoform_out
  RSEM_gene: File, outputSource: rsem/gene_out
  RNASeQC_Metrics: File, outputSource: rna_seqc/Metrics
  RNASeQC_counts: File, outputSource: supplemental/RNASeQC_counts # contains gene tpm, gene read, and exon counts
  kallisto_Abundance:  File, outputSource: kallisto/abundance_out
  annofuse_filtered_fusions_tsv:  &#39;File?&#39;, outputSource: annoFuse_filter/filtered_fusions_tsv, doc: &quot;Filtered output of formatted and annotated Star Fusion and arriba results&quot;
</pre></div>
</div>
<a class="reference external image-reference" href="https://github.com/kids-first/kf-rnaseq-workflow/blob/master/docs/kfdrc-rnaseq-workflow.png?raw=true"><img alt="pipeline flowchart" src="https://github.com/kids-first/kf-rnaseq-workflow/blob/master/docs/kfdrc-rnaseq-workflow.png?raw=true" /></a>
<section id="d3b-annofuse-workflow">
<h5>D3b annoFuse Workflow<a class="headerlink" href="#d3b-annofuse-workflow" title="Permalink to this headline">¶</a></h5>
<section id="introduction">
<h6>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h6>
<p>In this workflow, annoFuse performs standardization of StarFusion and arriba output files to retain information regarding fused genes, breakpoints, reading frame information as well as annotation from FusionAnnotator, output format description <a class="reference external" href="https://github.com/d3b-center/annoFuse/wiki#1-standardize-calls-from-fusion-callers-to-retain-information-regarding-fused-genesbreakpoints-reading-frame-information-as-well-as-annotation-from-fusionannotator">here</a>. Basic artifact filtering to remove fusions among gene paralogs, conjoined genes and fused genes found in normal samples is also performed by filtering fusions annotated by <a class="reference external" href="https://github.com/d3b-center/FusionAnnotator">FusionAnnotator</a> with “GTEx_Recurrent|DGD_PARALOGS|Normal|BodyMap|ConjoinG”. Each fusion call needs at least one junction reads support to be retained as true call. Additionally, if a fusion call has large number of spanning fragment reads compared to junction reads (spanning fragment minus junction read greater than ten), we remove these calls as potential false positives. An expression based filter is also applied, requiring a min FPKM value of 1 for the fusion genes in question.</p>
<p>Please refer to <a class="reference external" href="https://github.com/d3b-center/annoFuse">annoFuse</a> R package for additional applications like putative oncogene annotations.</p>
</section>
<section id="usage">
<h6>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h6>
<section id="inputs">
<h7>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h7>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span>inputs:
  sample_name:  &#39;string&#39;, doc: &quot;Sample name used for file base name of all outputs&quot;
  FusionGenome:  &#39;File&#39;, doc: &quot;GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz&quot;, sbg:suggestedValue: &#39;File&#39;, path: &#39;5d9c8d04e4b0950cce147f94&#39;, name: &#39;GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz&#39;
  genome_untar_path:   &#39;string?&#39;, doc: &quot;This is what the path will be when genome_tar is unpackaged&quot;, default: &quot;GRCh38_v27_CTAT_lib_Feb092018/ctat_genome_lib_build_dir&quot;
  rsem_expr_file: &#39;File&#39;, doc: &quot;gzipped rsem gene expression file&quot;
  arriba_output_file: &#39;File&#39;, doc: &quot;Output from arriba, usually extension arriba.fusions.tsv&quot;
  col_num:  &#39;int?&#39;, doc: &quot;column number in file of fusion name.&quot;
  star_fusion_output_file: &#39;File&#39;, doc: &quot;Output from arriba, usually extension STAR.fusion_predictions.abridged.coding_effect.tsv&quot;
  output_basename:  &#39;string&#39;, doc: &quot;String to use as basename for outputs&quot;
</pre></div>
</div>
</section>
<section id="id8">
<h7>Run<a class="headerlink" href="#id8" title="Permalink to this headline">¶</a></h7>
<ol class="arabic simple">
<li><p>Outputs from the arriba and STAR Fusion runs are required ahead of time (main RNAseq worflow output)</p></li>
<li><p>Gzipped rsem counts file, also generated in main RNAseq workflow</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">FusionGenome</span></code> should match what was used to run STAR Fusion</p></li>
</ol>
</section>
<section id="id9">
<h7>Outputs<a class="headerlink" href="#id9" title="Permalink to this headline">¶</a></h7>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">outputs</span><span class="p">:</span>
  <span class="nt">annofuse_filtered_fusions_tsv</span><span class="p">:</span>  <span class="l l-Scalar l-Scalar-Plain">File, outputSource</span><span class="p p-Indicator">:</span> <span class="l l-Scalar l-Scalar-Plain">annoFuse_filter/filtered_fusions_tsv, doc</span><span class="p p-Indicator">:</span> <span class="s">&quot;Filtred</span><span class="nv"> </span><span class="s">output</span><span class="nv"> </span><span class="s">of</span><span class="nv"> </span><span class="s">formatted</span><span class="nv"> </span><span class="s">and</span><span class="nv"> </span><span class="s">annotated</span><span class="nv"> </span><span class="s">Star</span><span class="nv"> </span><span class="s">Fusion</span><span class="nv"> </span><span class="s">and</span><span class="nv"> </span><span class="s">arriba</span><span class="nv"> </span><span class="s">results&quot;</span>
</pre></div>
</div>
</section>
</section>
</section>
</section>
</section>
<section id="id10">
<h3>Input parameters<a class="headerlink" href="#id10" title="Permalink to this headline">¶</a></h3>
<p><cite>sample_input.json</cite></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;gender&quot;</span><span class="p">:</span> <span class="s2">&quot;M&quot;</span><span class="p">,</span>
  <span class="s2">&quot;flowcell_id&quot;</span><span class="p">:</span> <span class="s2">&quot;JFHSKJHKJAH&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fastq1_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/kf_rnaseq_tests/SRR064286_sorted_chr20_45_50_sort.bam_R1.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fastq2_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/kf_rnaseq_tests/SRR064286_sorted_chr20_45_50_sort.bam_R2.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fusions_list&quot;</span><span class="p">:</span> <span class="s2">&quot;fusions.listA.txt&quot;</span><span class="p">,</span>
  <span class="s2">&quot;is_test&quot;</span><span class="p">:</span> <span class="s2">&quot;_tests&quot;</span><span class="p">,</span>
  <span class="s2">&quot;patient_id&quot;</span><span class="p">:</span> <span class="s2">&quot;rnaseq_test_PID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;phenotype&quot;</span><span class="p">:</span> <span class="s2">&quot;proband&quot;</span><span class="p">,</span>
  <span class="s2">&quot;sample_id&quot;</span><span class="p">:</span> <span class="s2">&quot;rnaseq_test_SID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;template_path&quot;</span><span class="p">:</span> <span class="s2">&quot;dags/cwl_kf_rnaseq/configurations/templates/kf_rnaseq_configuration_template.yaml&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="id11">
<h3>Sample REST API call<a class="headerlink" href="#id11" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -v -g -X POST <span class="se">\</span>
  http://localhost:8080/api/v1/dags/kf_rnaseq/dagRuns <span class="se">\</span>
  -H <span class="s1">&#39;Cache-Control: no-cache&#39;</span> <span class="se">\</span>
  -H <span class="s1">&#39;Content-Type: application/json&#39;</span> <span class="se">\</span>
  -d <span class="s2">&quot;{&#39;conf&#39;:\&#39;</span><span class="k">$(</span>cat sample_input.json <span class="p">|</span> sed <span class="s1">&#39;s/\&quot;/\\&quot;/g&#39;</span><span class="k">)</span><span class="s2">\&#39;}&quot;</span>
</pre></div>
</div>
</section>
<section id="id12">
<h3>Sample Airflow cli invocation<a class="headerlink" href="#id12" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>airflow dags trigger <span class="se">\</span>
-c <span class="s2">&quot;</span><span class="k">$(</span>cat sample_input.json<span class="k">)</span><span class="s2">&quot;</span> <span class="se">\</span>
-r dag_run-1 kf_rnaseq
</pre></div>
</div>
</section>
</section>
<section id="somatic-exome">
<h2>somatic-exome<a class="headerlink" href="#somatic-exome" title="Permalink to this headline">¶</a></h2>
<img alt="../_images/somatic-exome.png" src="../_images/somatic-exome.png" />
<section id="id13">
<h3>Description<a class="headerlink" href="#id13" title="Permalink to this headline">¶</a></h3>
<p>Somatic exome workflow based on <a class="reference external" href="https://github.com/sysamgen/analysis-workflows/blob/master/definitions/pipelines/somatic_exome.cwl">mgi-somatic-exome</a>.</p>
<p>Somatic_exome is designed to perform processing of mutant/wildtype H.sapiens
exome sequencing data. It features BQSR corrected alignments, 4 caller variant
detection, and vep style annotations. Structural variants are detected via
manta and cnvkit. In addition QC metrics are run, including
somalier concordance metrics.</p>
<p>Pipeline starts with either fastq files or unaligned BAM files with necessary metadata describing analysis type, sample name, gender, distinction between tumor and normal.</p>
<p>This step is performed seperately for normal and tumor sample.</p>
<img alt="../_images/alignment_qc1.png" src="../_images/alignment_qc1.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1093/bioinformatics/btp324">bwa_mem 0.7.15</a> - for sequence alignment step using GRCh38 reference</p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu314">samblaster 0.1.24</a> - for adding mate tags</p></li>
<li><p><a class="reference external" href="10.3390/biology1030895">flexbar 3.5</a> - for bar and adapter removal</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard:2.23.6</a> for alignment metrics</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btp352">samtools 1.11–h6270b1f_0</a> for BAM indexing and flagstat operation</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard 2.18.1</a> - for bedtointervals operation</p></li>
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk 4.1.8.1</a>  for Select Variants</p></li>
<li><p><a class="reference external" href="doi:10.1016/j.ajhg.2012.09.004">verifyBamId v1.1.3</a> - for checking whether the reads are contaminated as a mixture of two samples</p></li>
<li><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a> - for bam QC report</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btv566">qualimap</a> - for bam QC report</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btw354">multiqc</a> - QC</p></li>
</ul>
<p>This step is performed for normal and tumor sample at once.
Once variants are standardized they are annotated with metadata collected and published by public databases.
In the afterburner step final postprocessing is performed including generating URLs to external databases and extracting genotype information from INFO field.</p>
<img alt="../_images/detect_variants1.png" src="../_images/detect_variants1.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1038/nbt.2514">mutect 2</a> - for somatic variants detection</p></li>
<li><p><a class="reference external" href="10.1038/s41592-018-0051-x">strelka 2.9.9</a> - for somatic variants (SNPs and indels) detection</p></li>
<li><p><a class="reference external" href="10.1101/gr.129684.111">varscan 2.4.2</a> - for somatic variants detection</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btv710">manta 1.6.0</a> - for somatic structural variants detection</p></li>
<li><p><a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1004873">cnvkit 0.9.8</a> - for copy number variants detection</p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btv112">vt 0.57721–hf74b74d_1</a> - for normalization and decomposition</p></li>
<li><p><a class="reference external" href="10.1093/gigascience/giab008">bcftools 1.12</a> - for vcf manipulations (merging, renaming samples)</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btp352">samtools 1.11</a>  - for indexing and sanitizing BAM files (removing mutect-specific format tags, removing non ACTGN bases)</p></li>
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk 4.1.8.1</a> - for DOCM variants detection and selecting, filtering variants</p></li>
<li><p><a class="reference external" href="10.1093/gigascience/giab008">bcftools</a> - QC for detected variants</p></li>
</ul>
</section>
<section id="id27">
<h3>Input parameters<a class="headerlink" href="#id27" title="Permalink to this headline">¶</a></h3>
<p><cite>sample_input.json</cite></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;design&quot;</span><span class="p">:</span> <span class="s2">&quot;SSV6&quot;</span><span class="p">,</span>
  <span class="s2">&quot;flowcell_id&quot;</span><span class="p">:</span> <span class="s2">&quot;JHDAXCKJDLSK&quot;</span><span class="p">,</span>
  <span class="s2">&quot;gender&quot;</span><span class="p">:</span> <span class="s2">&quot;M&quot;</span><span class="p">,</span>
  <span class="s2">&quot;is_test&quot;</span><span class="p">:</span> <span class="s2">&quot;_tests&quot;</span><span class="p">,</span>
  <span class="s2">&quot;normal_fastq1_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/somatic-exome_tests/SRR7890845_chr17_test1_R1.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;normal_fastq2_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/somatic-exome_tests/SRR7890845_chr17_test1_R2.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;normal_sample_id&quot;</span><span class="p">:</span> <span class="s2">&quot;normal_test_SID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;patient_id&quot;</span><span class="p">:</span> <span class="s2">&quot;somatic_test_PID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;phenotype&quot;</span><span class="p">:</span> <span class="s2">&quot;proband&quot;</span><span class="p">,</span>
  <span class="s2">&quot;template_path&quot;</span><span class="p">:</span> <span class="s2">&quot;dags/cwl/configurations/templates/somatic_exome_template.yaml&quot;</span><span class="p">,</span>
  <span class="s2">&quot;tumor_fastq1_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/somatic-exome_tests/SRR7890844_chr17_test1_R1.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;tumor_fastq2_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/somatic-exome_tests/SRR7890844_chr17_test1_R2.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;tumor_sample_id&quot;</span><span class="p">:</span> <span class="s2">&quot;tumor_test_SID&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="id28">
<h3>Sample REST API call<a class="headerlink" href="#id28" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -v -g -X POST <span class="se">\</span>
  http://localhost:8080/api/v1/dags/somatic-exome/dagRuns <span class="se">\</span>
  -H <span class="s1">&#39;Cache-Control: no-cache&#39;</span> <span class="se">\</span>
  -H <span class="s1">&#39;Content-Type: application/json&#39;</span> <span class="se">\</span>
  -d <span class="s2">&quot;{&#39;conf&#39;:\&#39;</span><span class="k">$(</span>cat sample_input.json <span class="p">|</span> sed <span class="s1">&#39;s/\&quot;/\\&quot;/g&#39;</span><span class="k">)</span><span class="s2">\&#39;}&quot;</span>
</pre></div>
</div>
</section>
<section id="id29">
<h3>Sample Airflow cli invocation<a class="headerlink" href="#id29" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>airflow dags trigger <span class="se">\</span>
-c <span class="s2">&quot;</span><span class="k">$(</span>cat sample_input.json<span class="k">)</span><span class="s2">&quot;</span> <span class="se">\</span>
-r dag_run-1 somatic-exome
</pre></div>
</div>
</section>
</section>
<section id="tumor-only-exome">
<h2>tumor-only-exome<a class="headerlink" href="#tumor-only-exome" title="Permalink to this headline">¶</a></h2>
<img alt="../_images/tumor-only-exome.png" src="../_images/tumor-only-exome.png" />
<section id="id30">
<h3>Description<a class="headerlink" href="#id30" title="Permalink to this headline">¶</a></h3>
<p>Tumor only exome workflow based on <a class="reference external" href="https://github.com/genome/analysis-workflows/blob/master/definitions/pipelines/tumor_only_exome.cwl">mgi-tumor-only-exome</a>.</p>
<p>Tumor_only_exome is designed to perform processing of mutant/wildtype H.sapiens
exome sequencing data. It features BQSR corrected alignments, varscan variant
detection, and vep style annotations. In addition QC metrics are run.</p>
<p>Pipeline starts with either fastq files or unaligned BAM files with necessary metadata describing analysis type, sample name, gender, distinction between tumor and normal.</p>
<p>This step is performed seperately for normal and tumor sample.</p>
<img alt="../_images/alignment_qc1.png" src="../_images/alignment_qc1.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1093/bioinformatics/btp324">bwa_mem 0.7.15</a> - for sequence alignment step using GRCh38 reference</p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu314">samblaster 0.1.24</a> - for adding mate tags</p></li>
<li><p><a class="reference external" href="10.3390/biology1030895">flexbar 3.5</a> - for bar and adapter removal</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard:2.23.6</a> for alignment metrics</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btp352">samtools 1.11–h6270b1f_0</a> for BAM indexing and flagstat operation</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard 2.18.1</a> - for bedtointervals operation</p></li>
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk 4.1.8.1</a>  for Select Variants</p></li>
<li><p><a class="reference external" href="doi:10.1016/j.ajhg.2012.09.004">verifyBamId v1.1.3</a> - for checking whether the reads are contaminated as a mixture of two samples</p></li>
<li><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a> - for bam QC report</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btv566">qualimap</a> - for bam QC report</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btw354">multiqc</a> - QC</p></li>
</ul>
<p>This step is performed for tumor sample only.</p>
<img alt="../_images/detect_variants1.png" src="../_images/detect_variants1.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1101/gr.129684.111">varscan 2.4.2</a> - for somatic variants detection</p></li>
<li><p><a href="#id51"><span class="problematic" id="id52">`cle 1.4.2 &lt;&gt;`_</span></a></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btv112">vt 0.57721–hf74b74d_1</a> - for normalization and decomposition</p></li>
<li><p><a class="reference external" href="10.1093/gigascience/giab008">bcftools 1.12</a> - for vcf manipulations (merging, renaming samples)</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btp352">samtools 1.11</a>  - for indexing and sanitizing BAM files (removing mutect-specific format tags, removing non ACTGN bases)</p></li>
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk 4.1.8.1</a> - for DOCM variants detection and selecting, filtering variants</p></li>
<li><p><a class="reference external" href="10.1093/gigascience/giab008">bcftools</a> - QC for detected variants</p></li>
</ul>
</section>
<section id="id48">
<h3>Input parameters<a class="headerlink" href="#id48" title="Permalink to this headline">¶</a></h3>
<p><cite>sample_input.json</cite></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;design&quot;</span><span class="p">:</span> <span class="s2">&quot;SSV6&quot;</span><span class="p">,</span>
  <span class="s2">&quot;flowcell_id&quot;</span><span class="p">:</span> <span class="s2">&quot;JHDAXCKJDLSK&quot;</span><span class="p">,</span>
  <span class="s2">&quot;gender&quot;</span><span class="p">:</span> <span class="s2">&quot;M&quot;</span><span class="p">,</span>
  <span class="s2">&quot;is_test&quot;</span><span class="p">:</span> <span class="s2">&quot;_tests&quot;</span><span class="p">,</span>
  <span class="s2">&quot;patient_id&quot;</span><span class="p">:</span> <span class="s2">&quot;tumor_only_test_PID&quot;</span><span class="p">,</span>
  <span class="s2">&quot;phenotype&quot;</span><span class="p">:</span> <span class="s2">&quot;proband&quot;</span><span class="p">,</span>
  <span class="s2">&quot;template_path&quot;</span><span class="p">:</span> <span class="s2">&quot;dags/cwl/configurations/templates/tumor_only_exome_template.yaml&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fastq1_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/somatic-exome_tests/SRR7890844_chr17_test1_R1.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;fastq2_path&quot;</span><span class="p">:</span> <span class="s2">&quot;cwl_inputs_folder/somatic-exome_tests/SRR7890844_chr17_test1_R2.fastq.gz&quot;</span><span class="p">,</span>
  <span class="s2">&quot;sample_id&quot;</span><span class="p">:</span> <span class="s2">&quot;tumor_only_test_SID&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="id49">
<h3>Sample REST API call<a class="headerlink" href="#id49" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -v -g -X POST <span class="se">\</span>
  http://localhost:8080/api/v1/dags/tumor-only-exome/dagRuns <span class="se">\</span>
  -H <span class="s1">&#39;Cache-Control: no-cache&#39;</span> <span class="se">\</span>
  -H <span class="s1">&#39;Content-Type: application/json&#39;</span> <span class="se">\</span>
  -d <span class="s2">&quot;{&#39;conf&#39;:\&#39;</span><span class="k">$(</span>cat sample_input.json <span class="p">|</span> sed <span class="s1">&#39;s/\&quot;/\\&quot;/g&#39;</span><span class="k">)</span><span class="s2">\&#39;}&quot;</span>
</pre></div>
</div>
</section>
<section id="id50">
<h3>Sample Airflow cli invocation<a class="headerlink" href="#id50" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>airflow dags trigger <span class="se">\</span>
-c <span class="s2">&quot;</span><span class="k">$(</span>cat sample_input.json<span class="k">)</span><span class="s2">&quot;</span> <span class="se">\</span>
-r dag_run-1 tumor-only-exome
</pre></div>
</div>
</section>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">samgen</a></h1>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../overview/overview.html">System overview</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Pipelines overview</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#aux-render-notebook">aux-render-notebook</a></li>
<li class="toctree-l2"><a class="reference internal" href="#germline-exome">germline-exome</a></li>
<li class="toctree-l2"><a class="reference internal" href="#kf-rnaseq">kf_rnaseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="#somatic-exome">somatic-exome</a></li>
<li class="toctree-l2"><a class="reference internal" href="#tumor-only-exome">tumor-only-exome</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../tech_docs/tech_docs.html">Running pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../catalogue/catalogue.html">Data catalogue</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="../overview/overview.html" title="previous chapter">System overview</a></li>
      <li>Next: <a href="../tech_docs/tech_docs.html" title="next chapter">Running pipelines</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, Agnieszka Szmurło, Tomasz Gambin, Mateusz Dawidziuk, Dominika Maziec, Marek Wiewiórka.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../_sources/dags/dags.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>